app/views/play/OzariaUnitMapLevelDot.vue

```diff
@@ -50,13 +50,17 @@ export default Vue.extend({
       if (this.levelData.locked)
         return '#'
 
+      if (this.levelData.type == 'intro') {
+        link = '/ozaria/play/intro/' + this.levelData.slug
+      } else {
+        link = '/ozaria/play/level/' + this.levelData.slug
+      }
+
       if (this.courseId && this.courseInstanceId) {
-        link = "/play/level/"+this.levelData.slug+"?course="+this.courseId+"&course-instance="+this.courseInstanceId
+        link += "?course=" + this.courseId + "&course-instance=" + this.courseInstanceId
         if (this.levelData.primerLanguage)
           link += "&codeLanguage=" + this.levelData.primerLanguage 
```

Comment:
Idea: given this URL construction is a little more complicated it could be done with fetch's recommended query string construction approach (which handles encoding for you).

https://fetch.spec.whatwg.org/#fetch-blob-example
4th example down.

Reply:
Those are great suggestions, but here the link is not being used for an HTTP request, instead, it is used to route the application to a new URL. I will use ES6 string interpolation and encodeURIComponent.